doc/usecases/record_elektra/UC_change_notifications.md

```diff
@@ -0,0 +1,23 @@
+# Use Case: Notify plugins when data in the KDB changes
+
+## Summary
+
+- **Title:** Notify plugins when data in the KDB changes
+- **Scope:** Configuration/Logging
+- **Level:** Developer goal
+- **Actors:** Plugin developer
+- **Brief:** It should be possible for plugins to be notified when data in KDB changes.
```

Comment:
IMO this is to general. To me this reads like: "A plugin whose main purpose is something else (e.g. `type`) can be notified of changes via an extra API." I think that would lead to chaos.

IMO what we should have is:

> Some plugins are only executed when something changed in the KDB (*) and as an input receive the keys that changed in their old and new state.

Importantly, the plugins should already receive the changeset as an input. In other words, `libelektra-kdb` (or an extra component between `libelektra-kdb` and the plugins) does the change tracking not the plugins themselves. This is important, because otherwise every such "change plugin" would keep its own copy of all keys to keep track of the old values.

---

(*) This is already the case for standard mounted backend plugins. They are only called, if there is a changed key in their mountpoint. But they always receive all keys for the mountpoint and don't receive the old state.

Reply:
I agree, we need this narrowed down, maybe even several UC (logging, change notification, ...)

Reply:
> logging, change notification

I don't think that separation would make sense, since "logging" is an application/special case of "change notification". The plugin is notified of changes and just logs them somewhere, instead of doing something else (e.g. broadcasting via dbus)

However, it might make sense to make one UC for "global/hook plugins" in general and a second UC for a "logging plugin". But this may be a too technical distinction, as "global/hook plugins" could just be considered the implementation details of the current UC.

Reply:
>  log read-only accesses to the kdb, maybe for audit purposes

Yeah, I think this would be very far out of scope. Logging read accesses a totally different can of worms.

What should be possible with the basic logging/notification hook is logging the parent key that was used in every `kdbGet`. Any logging/notification plugin would be called with the parent key anyway.